libs/core/routing/src/query-param-filter/request-builder/type.ts

```diff
@@ -0,0 +1,8 @@
+import { ParamMap } from '@angular/router';
+
+import { DaffFilterRequest } from '@daffodil/core';
+
+/**
+ * Accepts a list of values set to a particular URL query param and returns a filter request.
+ */
+export type DaffRoutingQueryParamFilterRequestBuilder<T extends DaffFilterRequest = DaffFilterRequest> = (qpValue: ReturnType<ParamMap['getAll']>, filterName: string) => T;
```

Comment:
I think the `queryParam` key should be the first arg in the function arguments of this type.